File: download-service-go/internal/delivery/dto/video_info_dto.go
package dto

type VideoInfoMessageDto struct {
	Pattern string       `json:"pattern"`
	Data    VideoInfoDto `json:"data"`
}

type VideoInfoDto struct {
	VideoName   string `json:"videoName"`
	FolderID    string `json:"folderID"`
	RealPath    string `json:"realPath"`
	PreviewPath string `json:"previewPath"`
}

File: management-service-nodejs/src/modules/videos/dto/video-info.dto.ts
import { IsNotEmpty } from 'class-validator';
import { Types } from 'mongoose';
import { IsObjectId } from '../../../lib/decorators/isObjectID.decorator';
import { Transform } from 'class-transformer';

export class VideoInfoDto {
  @IsNotEmpty()
  videoName: string;

  @IsNotEmpty()
  @IsObjectId({ message: 'folderID should be objectID' })
  @Transform(({ value }) =>
    value && Types.ObjectId.isValid(value as string)
      ? new Types.ObjectId(value as string)
      : value,
  )
  folderID: Types.ObjectId;

  @IsNotEmpty()
  realPath: string;

  @IsNotEmpty()
  previewPath: string;
}
